// react-rainbow-components/components/ImportRecordsFlow/stepTwo/preview.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = Preview;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _ButtonIcon = _interopRequireDefault(require("../../ButtonIcon"));

var _Table = _interopRequireDefault(require("../../Table"));

var _Column = _interopRequireDefault(require("../../Column"));

var _Spinner = _interopRequireDefault(require("../../Spinner"));

var _csv = _interopRequireDefault(require("../icons/csv"));

var _trash = _interopRequireDefault(require("../icons/trash"));

function PreviewTable(props) {
  var columns = props.columns,
      data = props.data,
      isLoading = props.isLoading;

  if (isLoading) {
    return _react["default"].createElement("div", {
      className: "rainbow-import-records-flow_step-two-spinner-container"
    }, _react["default"].createElement(_Spinner["default"], null));
  }

  return _react["default"].createElement(_Table["default"], {
    className: "rainbow-import-records-flow_table",
    keyField: "id",
    data: data
  }, columns.map(function (col) {
    return _react["default"].createElement(_Column["default"], {
      key: col,
      header: col,
      field: col
    });
  }));
}

PreviewTable.propTypes = {
  columns: _propTypes["default"].array.isRequired,
  data: _propTypes["default"].array.isRequired,
  isLoading: _propTypes["default"].bool.isRequired
};

function Preview(props) {
  var fileName = props.fileName,
      fileType = props.fileType,
      columns = props.columns,
      data = props.data,
      isLoading = props.isLoading,
      onRemoveFile = props.onRemoveFile;
  var previewData = data.slice(0, 5);
  return _react["default"].createElement("div", null, _react["default"].createElement("div", {
    className: "rainbow-import-records-flow_step-two-file-card"
  }, _react["default"].createElement(_csv["default"], null), _react["default"].createElement("div", {
    className: "rainbow-import-records-flow_step-two-file-card-content"
  }, _react["default"].createElement("h1", {
    title: fileName,
    className: "rainbow-import-records-flow_step-two-file-card-title"
  }, fileName), _react["default"].createElement("h2", {
    className: "rainbow-import-records-flow_step-two-file-card-description"
  }, fileType)), _react["default"].createElement(_ButtonIcon["default"], {
    className: "rainbow-import-records-flow_step-two-file-card-delete-button-icon",
    icon: _react["default"].createElement(_trash["default"], {
      className: "rainbow-import-records-flow_step-two-file-card-delete-icon"
    }),
    title: "Remove file",
    assistiveText: "Remove file",
    onClick: onRemoveFile
  })), _react["default"].createElement(PreviewTable, {
    columns: columns,
    data: previewData,
    isLoading: isLoading
  }));
}

Preview.propTypes = {
  fileName: _propTypes["default"].string,
  fileType: _propTypes["default"].string,
  columns: _propTypes["default"].array,
  data: _propTypes["default"].array,
  isLoading: _propTypes["default"].bool,
  onRemoveFile: _propTypes["default"].func
};
Preview.defaultProps = {
  fileName: '',
  fileType: '',
  columns: [],
  data: [],
  isLoading: false,
  onRemoveFile: function onRemoveFile() {}
};
