// react-rainbow-components/components/ImportRecordsFlow/stepTwo/uploadFileButton.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = UploadFileButton;

var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));

var _react = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _upload = _interopRequireDefault(require("../icons/upload"));

var _utils = require("../../../libs/utils");

function UploadFileButton(_ref) {
  var onChange = _ref.onChange;

  var _useState = (0, _react.useState)((0, _utils.uniqueId)('upload-file-input')),
      _useState2 = (0, _slicedToArray2["default"])(_useState, 1),
      uploadFileInputId = _useState2[0];

  return _react["default"].createElement("label", {
    className: "rainbow-import-records-flow_step-two_upload-file-label",
    htmlFor: uploadFileInputId
  }, _react["default"].createElement("input", {
    id: uploadFileInputId,
    className: "rainbow-import-records-flow_step-two_upload-file-input",
    type: "file",
    accept: "text/csv",
    multiple: false,
    onChange: onChange
  }), _react["default"].createElement(_upload["default"], {
    className: "rainbow-import-records-flow_step-two-upload-button-icon"
  }), "Upload CSV File");
}

UploadFileButton.propTypes = {
  onChange: _propTypes["default"].func.isRequired
};
