// react-rainbow-components/components/Slider/label.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = Label;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _classnames = _interopRequireDefault(require("classnames"));

function Label(props) {
  var label = props.label,
      sliderId = props.sliderId,
      hideLabel = props.hideLabel;

  var getLabelClassNames = function getLabelClassNames() {
    return (0, _classnames["default"])('rainbow-slider_label', {
      'rainbow-slider_label--hide-label': hideLabel
    });
  };

  return _react["default"].createElement("label", {
    className: getLabelClassNames(),
    htmlFor: sliderId
  }, label);
}

Label.propTypes = {
  label: _propTypes["default"].node.isRequired,
  sliderId: _propTypes["default"].string.isRequired,
  hideLabel: _propTypes["default"].bool
};
Label.defaultProps = {
  hideLabel: false
};
