// react-rainbow-components/components/Breadcrumb/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = Breadcrumb;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _classnames = _interopRequireDefault(require("classnames"));

require("./styles.css");

/* eslint-disable no-script-url */

/**
 * An item in the hierarchy path of the page the user is on.
 * @category Layout
 */
function Breadcrumb(props) {
  var href = props.href,
      label = props.label,
      onClick = props.onClick,
      disabled = props.disabled,
      className = props.className,
      style = props.style;

  var getClassNames = function getClassNames() {
    return (0, _classnames["default"])('rainbow-breadcrumb', {
      'rainbow-breadcrumb--disabled': disabled
    }, className);
  };

  return _react["default"].createElement("li", {
    className: getClassNames(),
    style: style
  }, _react["default"].createElement("a", {
    className: "rainbow-breadcrumb-anchor",
    href: href,
    onClick: onClick,
    "aria-disabled": !!disabled
  }, label));
}

Breadcrumb.propTypes = {
  /** The text label for the breadcrumb. */
  label: _propTypes["default"].oneOfType([_propTypes["default"].string, _propTypes["default"].node]).isRequired,

  /** The URL of the page that the breadcrumb goes to. */
  href: _propTypes["default"].string,

  /** The action triggered when the breadcrumb is clicked. */
  onClick: _propTypes["default"].func,

  /** Specifies that a breadcrumb element should be disabled. This value defaults to false. */
  disabled: _propTypes["default"].bool,

  /** A CSS class for the outer element, in addition to the component's base classes. */
  className: _propTypes["default"].string,

  /** An object with custom style applied to the outer element. */
  style: _propTypes["default"].object
};
Breadcrumb.defaultProps = {
  href: 'javascript:void(0);',
  onClick: function onClick() {},
  disabled: false,
  className: undefined,
  style: undefined
};
