// react-rainbow-components/components/Pagination/pageButtons.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = PageButtons;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _classnames = _interopRequireDefault(require("classnames"));

var _getFirstItem = _interopRequireDefault(require("./getFirstItem"));

/* eslint-disable no-script-url */
function PageButtons(props) {
  var pages = props.pages,
      activePage = props.activePage,
      onChange = props.onChange;

  var getButtonClassName = function getButtonClassName(page) {
    return (0, _classnames["default"])('rainbow-pagination_button-content', {
      'rainbow-pagination_button-content--active': activePage === page
    });
  };

  var getAriaCurrent = function getAriaCurrent(page) {
    if (page === activePage) {
      return 'page';
    }

    return undefined;
  };

  var renderButtons = function renderButtons() {
    var firstItem = pages > 4 ? (0, _getFirstItem["default"])(pages, activePage) : 1;
    var buttonsToRender = pages > 4 ? 5 : pages;
    return Array(buttonsToRender).fill(0).map(function (item, index) {
      var page = firstItem + index;
      var key = "page-button-".concat(page);
      var ariaLabel = "Goto Page ".concat(page);
      return _react["default"].createElement("li", {
        key: key,
        className: "rainbow-pagination_button"
      }, _react["default"].createElement("a", {
        className: getButtonClassName(page),
        onClick: function onClick(event) {
          return onChange(event, page);
        },
        "aria-current": getAriaCurrent(page),
        "aria-label": ariaLabel,
        href: "javascript:void(0);"
      }, _react["default"].createElement("span", null, page)));
    });
  };

  return renderButtons();
}

PageButtons.propTypes = {
  activePage: _propTypes["default"].number,
  pages: _propTypes["default"].number.isRequired,
  onChange: _propTypes["default"].func.isRequired
};
PageButtons.defaultProps = {
  activePage: undefined
};
