// react-rainbow-components/components/TimePicker/input/index.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _react = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _classnames2 = _interopRequireDefault(require("classnames"));

var _utils = require("./../../../libs/utils");

var _label = _interopRequireDefault(require("./label"));

var _RenderIf = _interopRequireDefault(require("../../RenderIf"));

require("./styles.css");

var Input =
/*#__PURE__*/
function (_Component) {
  (0, _inherits2["default"])(Input, _Component);

  function Input(props) {
    var _this;

    (0, _classCallCheck2["default"])(this, Input);
    _this = (0, _possibleConstructorReturn2["default"])(this, (0, _getPrototypeOf2["default"])(Input).call(this, props));
    _this.inputId = (0, _utils.uniqueId)('input');
    _this.inlineTextLabelId = (0, _utils.uniqueId)('inline-text-label');
    _this.errorMessageId = (0, _utils.uniqueId)('error-message');
    _this.inputRef = _react["default"].createRef();
    return _this;
  }

  (0, _createClass2["default"])(Input, [{
    key: "getContainerClassNames",
    value: function getContainerClassNames() {
      var _this$props = this.props,
          className = _this$props.className,
          error = _this$props.error;
      return (0, _classnames2["default"])('rainbow-time-picker_input-container', {
        'rainbow-time-picker_input--error': error
      }, className);
    }
  }, {
    key: "getIconPositionClassNames",
    value: function getIconPositionClassNames() {
      var _this$props2 = this.props,
          icon = _this$props2.icon,
          iconPosition = _this$props2.iconPosition;
      return (0, _classnames2["default"])((0, _defineProperty2["default"])({
        'rainbow-time-picker_input-icon-container': icon
      }, "rainbow-time-picker_input-icon--".concat(iconPosition), icon));
    }
  }, {
    key: "getInputClassNames",
    value: function getInputClassNames() {
      var _this$props3 = this.props,
          isBare = _this$props3.isBare,
          isCentered = _this$props3.isCentered,
          readOnly = _this$props3.readOnly;
      return (0, _classnames2["default"])('rainbow-time-picker_input', {
        'rainbow-time-picker_input--bare': isBare,
        'rainbow-time-picker_input--counter': isCentered,
        'rainbow-time-picker_input--readonly': readOnly
      });
    }
  }, {
    key: "getInlineTextLabelId",
    value: function getInlineTextLabelId() {
      var bottomHelpText = this.props.bottomHelpText;

      if (bottomHelpText) {
        return this.inlineTextLabelId;
      }

      return undefined;
    }
  }, {
    key: "getErrorMessageId",
    value: function getErrorMessageId() {
      var error = this.props.error;

      if (error) {
        return this.errorMessageId;
      }

      return undefined;
    }
    /**
     * Sets focus on the element.
     * @public
     */

  }, {
    key: "focus",
    value: function focus() {
      this.inputRef.current.focus();
    }
    /**
     * Sets click on the element.
     * @public
     */

  }, {
    key: "click",
    value: function click() {
      this.inputRef.current.click();
    }
    /**
     * Sets blur on the element.
     * @public
     */

  }, {
    key: "blur",
    value: function blur() {
      this.inputRef.current.blur();
    }
  }, {
    key: "render",
    value: function render() {
      var _this$props4 = this.props,
          style = _this$props4.style,
          value = _this$props4.value,
          onChange = _this$props4.onChange,
          label = _this$props4.label,
          error = _this$props4.error,
          placeholder = _this$props4.placeholder,
          disabled = _this$props4.disabled,
          readOnly = _this$props4.readOnly,
          tabIndex = _this$props4.tabIndex,
          onFocus = _this$props4.onFocus,
          onBlur = _this$props4.onBlur,
          onClick = _this$props4.onClick,
          onKeyDown = _this$props4.onKeyDown,
          type = _this$props4.type,
          maxLength = _this$props4.maxLength,
          minLength = _this$props4.minLength,
          pattern = _this$props4.pattern,
          icon = _this$props4.icon,
          bottomHelpText = _this$props4.bottomHelpText,
          required = _this$props4.required,
          id = _this$props4.id,
          autoComplete = _this$props4.autoComplete,
          name = _this$props4.name,
          hideLabel = _this$props4.hideLabel;
      return _react["default"].createElement("div", {
        id: id,
        className: this.getContainerClassNames(),
        style: style
      }, _react["default"].createElement(_label["default"], {
        label: label,
        hideLabel: hideLabel,
        required: required,
        inputId: this.inputId,
        readOnly: readOnly,
        id: this.getInlineTextLabelId()
      }), _react["default"].createElement("div", {
        className: this.getIconPositionClassNames()
      }, _react["default"].createElement(_RenderIf["default"], {
        isTrue: !!icon
      }, _react["default"].createElement("span", {
        className: "rainbow-time-picker_input-icon"
      }, icon)), _react["default"].createElement("input", {
        id: this.inputId,
        name: name,
        type: type,
        className: this.getInputClassNames(),
        value: value,
        placeholder: placeholder,
        onChange: onChange,
        tabIndex: tabIndex,
        onFocus: onFocus,
        onBlur: onBlur,
        onClick: onClick,
        onKeyDown: onKeyDown,
        disabled: disabled,
        readOnly: true,
        required: required,
        maxLength: maxLength,
        minLength: minLength,
        pattern: pattern,
        autoComplete: autoComplete,
        "aria-labelledby": this.getInlineTextLabelId(),
        "aria-describedby": this.getErrorMessageId(),
        ref: this.inputRef
      })), _react["default"].createElement(_RenderIf["default"], {
        isTrue: !!bottomHelpText
      }, _react["default"].createElement("div", {
        className: "rainbow-time-picker_input--help"
      }, bottomHelpText)), _react["default"].createElement(_RenderIf["default"], {
        isTrue: !!error
      }, _react["default"].createElement("div", {
        id: this.getErrorMessageId(),
        className: "rainbow-time-picker_input-error"
      }, error)));
    }
  }]);
  return Input;
}(_react.Component);

exports["default"] = Input;
Input.propTypes = {
  value: _propTypes["default"].string,
  name: _propTypes["default"].string,
  type: _propTypes["default"].oneOf(['text', 'password', 'datetime', 'datetime-local', 'date', 'month', 'time', 'week', 'number', 'email', 'url', 'search', 'tel', 'color']),
  label: _propTypes["default"].oneOfType([_propTypes["default"].string, _propTypes["default"].node]).isRequired,
  placeholder: _propTypes["default"].string,
  icon: _propTypes["default"].node,
  iconPosition: _propTypes["default"].oneOf(['left', 'right']),
  maxLength: _propTypes["default"].number,
  minLength: _propTypes["default"].number,
  bottomHelpText: _propTypes["default"].oneOfType([_propTypes["default"].string, _propTypes["default"].node]),
  required: _propTypes["default"].bool,
  pattern: _propTypes["default"].string,
  isCentered: _propTypes["default"].bool,
  isBare: _propTypes["default"].bool,
  error: _propTypes["default"].oneOfType([_propTypes["default"].string, _propTypes["default"].node]),
  disabled: _propTypes["default"].bool,
  readOnly: _propTypes["default"].bool,
  onChange: _propTypes["default"].func,
  tabIndex: _propTypes["default"].oneOfType([_propTypes["default"].number, _propTypes["default"].string]),
  onClick: _propTypes["default"].func,
  onFocus: _propTypes["default"].func,
  onBlur: _propTypes["default"].func,
  onKeyDown: _propTypes["default"].func,
  className: _propTypes["default"].string,
  style: _propTypes["default"].object,
  id: _propTypes["default"].string,
  autoComplete: _propTypes["default"].string,
  hideLabel: _propTypes["default"].bool
};
Input.defaultProps = {
  value: undefined,
  name: undefined,
  type: 'text',
  placeholder: null,
  icon: undefined,
  iconPosition: 'left',
  maxLength: undefined,
  minLength: undefined,
  bottomHelpText: null,
  required: false,
  pattern: undefined,
  isCentered: false,
  isBare: false,
  error: null,
  disabled: false,
  readOnly: false,
  onChange: function onChange() {},
  tabIndex: undefined,
  onClick: function onClick() {},
  onFocus: function onFocus() {},
  onBlur: function onBlur() {},
  onKeyDown: function onKeyDown() {},
  className: undefined,
  style: undefined,
  id: undefined,
  autoComplete: 'on',
  hideLabel: false
};
