// react-rainbow-components/components/ProgressStep/index.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = ProgressStep;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _assertThisInitialized2 = _interopRequireDefault(require("@babel/runtime/helpers/assertThisInitialized"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _react = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _classnames = _interopRequireDefault(require("classnames"));

var _ButtonIcon = _interopRequireDefault(require("../ButtonIcon"));

var _context = require("../ProgressIndicator/context");

var _doneIcon = _interopRequireDefault(require("./doneIcon"));

var _errorIcon = _interopRequireDefault(require("./errorIcon"));

require("./styles.css");

/* eslint-disable react/prop-types */
var iconMap = {
  Error: function Error() {
    return _react["default"].createElement(_errorIcon["default"], null);
  },
  Completed: function Completed() {
    return _react["default"].createElement(_doneIcon["default"], null);
  },
  Active: function Active() {
    return _react["default"].createElement("div", {
      className: "rainbow-progress-step--is-active_icon"
    });
  },
  Inactive: function Inactive() {
    return _react["default"].createElement("div", {
      className: "rainbow-progress-step_marker_icon"
    });
  }
};

var StepItem =
/*#__PURE__*/
function (_Component) {
  (0, _inherits2["default"])(StepItem, _Component);

  function StepItem(props) {
    var _this;

    (0, _classCallCheck2["default"])(this, StepItem);
    _this = (0, _possibleConstructorReturn2["default"])(this, (0, _getPrototypeOf2["default"])(StepItem).call(this, props));
    _this.handleOnClick = _this.handleOnClick.bind((0, _assertThisInitialized2["default"])(_this));
    _this.setStepState = _this.setStepState.bind((0, _assertThisInitialized2["default"])(_this));
    _this.state = {
      stepState: 'Inactive'
    };
    return _this;
  }

  (0, _createClass2["default"])(StepItem, [{
    key: "componentDidMount",
    value: function componentDidMount() {
      var _this2 = this;

      var _this$props = this.props,
          privateRegisterStep = _this$props.privateRegisterStep,
          name = _this$props.name;
      return setTimeout(function () {
        return privateRegisterStep({
          name: name,
          onSetStepState: _this2.setStepState
        });
      }, 0);
    }
  }, {
    key: "componentDidUpdate",
    value: function componentDidUpdate(_ref) {
      var prevCurrentStepName = _ref.currentStepName;
      var _this$props2 = this.props,
          currentStepName = _this$props2.currentStepName,
          name = _this$props2.name;

      if (prevCurrentStepName !== currentStepName) {
        this.props.setChildrenState({
          name: name,
          onSetStepState: this.setStepState
        });
      }
    }
  }, {
    key: "setStepState",
    value: function setStepState(stepState) {
      var hasError = this.props.hasError;

      if (hasError) {
        return this.setState({
          stepState: 'Error'
        });
      }

      return this.setState({
        stepState: stepState
      });
    }
  }, {
    key: "getContainerClassNames",
    value: function getContainerClassNames() {
      var className = this.props.className;
      return (0, _classnames["default"])('rainbow-progress-step', className);
    }
  }, {
    key: "getButtonClassNames",
    value: function getButtonClassNames() {
      var stepState = this.state.stepState;
      return (0, _classnames["default"])({
        'rainbow-progress-step_marker': stepState === 'Inactive',
        'rainbow-progress-step--is-completed': stepState === 'Completed',
        'rainbow-progress-step--is-active': stepState === 'Active',
        'rainbow-progress-step--error': stepState === 'Error'
      });
    }
  }, {
    key: "getAssistiveText",
    value: function getAssistiveText() {
      var label = this.props.label;
      var stepState = this.state.stepState;

      if (label && stepState !== 'Inactive') {
        return "".concat(label, " - ").concat(stepState);
      }

      if (label) {
        return label;
      }

      if (stepState !== 'Inactive') {
        return stepState;
      }

      return '';
    }
  }, {
    key: "getIcon",
    value: function getIcon() {
      var stepState = this.state.stepState;

      if (iconMap[stepState]) {
        return iconMap[stepState]();
      }

      return null;
    }
  }, {
    key: "handleOnClick",
    value: function handleOnClick(event) {
      var _this$props3 = this.props,
          privateOnClick = _this$props3.privateOnClick,
          name = _this$props3.name;
      return privateOnClick(event, name);
    }
  }, {
    key: "render",
    value: function render() {
      var label = this.props.label;
      return _react["default"].createElement("li", {
        className: this.getContainerClassNames()
      }, _react["default"].createElement(_ButtonIcon["default"], {
        icon: this.getIcon(),
        className: this.getButtonClassNames(),
        onClick: this.handleOnClick,
        assistiveText: this.getAssistiveText()
      }), _react["default"].createElement("span", {
        className: "rainbow-progress-step_label"
      }, label));
    }
  }]);
  return StepItem;
}(_react.Component);

function ProgressStep(props) {
  return _react["default"].createElement(_context.Consumer, null, function (context) {
    return _react["default"].createElement(StepItem, (0, _extends2["default"])({}, props, context));
  });
}

ProgressStep.propTypes = {
  /** The name is used to determine which ProgressStep is active. */
  name: _propTypes["default"].string,

  /** Text label for the ProgressStep. */
  label: _propTypes["default"].node,

  /** Indicates whether the current step is in error state and displays a warning icon
   * on the step indicator. This value defaults to false. */
  hasError: _propTypes["default"].bool,

  /** A CSS class for the outer element, in addition to the component's base classes. */
  className: _propTypes["default"].string,

  /** An object with custom style applied to the outer element. */
  style: _propTypes["default"].object
};
ProgressStep.defaultProps = {
  name: '',
  label: '',
  hasError: false,
  className: undefined,
  style: undefined
};
