// react-rainbow-components/components/ProgressCircular/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = ProgressCircular;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _classnames = _interopRequireDefault(require("classnames"));

var _AssistiveText = _interopRequireDefault(require("./../AssistiveText"));

var _normalizeValue = _interopRequireDefault(require("./normalizeValue"));

var _progressRing = _interopRequireDefault(require("./progressRing"));

require("./styles.css");

/**
 * ProgressCircular component communicates to the user the progress of a particular process.
 */
function ProgressCircular(props) {
  var value = props.value,
      variant = props.variant,
      assistiveText = props.assistiveText,
      className = props.className,
      style = props.style;

  var getContainerClassNames = function getContainerClassNames() {
    return (0, _classnames["default"])('rainbow-progress-circular', {
      'rainbow-progress-circular--success': variant === 'success',
      'rainbow-progress-circular--warning': variant === 'warning',
      'rainbow-progress-circular--error': variant === 'error'
    }, className);
  };

  var normalizedValue = (0, _normalizeValue["default"])(value);
  return _react["default"].createElement("div", {
    className: getContainerClassNames(),
    "aria-valuemin": "0",
    "aria-valuemax": "100",
    "aria-valuenow": normalizedValue,
    role: "progressbar",
    style: style
  }, _react["default"].createElement(_progressRing["default"], {
    percent: normalizedValue
  }), _react["default"].createElement("span", {
    className: "rainbow-progress-circular_percent-text"
  }, "".concat(normalizedValue, "%")), _react["default"].createElement(_AssistiveText["default"], {
    text: assistiveText
  }));
}

ProgressCircular.propTypes = {
  /** The percentage value of the progress circular. It must be between 0 and 100. */
  value: _propTypes["default"].number,

  /** The variant of the progress circular. Valid values are brand and success. */
  variant: _propTypes["default"].oneOf(['brand', 'success', 'warning', 'error']),

  /** A description for assistive sreen readers. */
  assistiveText: _propTypes["default"].string,

  /** A CSS class for the outer element, in addition to the component's base classes. */
  className: _propTypes["default"].string,

  /** An object with custom style applied to the outer element. */
  style: _propTypes["default"].object
};
ProgressCircular.defaultProps = {
  value: 0,
  variant: 'brand',
  assistiveText: '',
  className: undefined,
  style: undefined
};
