// react-rainbow-components/components/ProgressCircular/progressRing.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = ProgressRing;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

function ProgressRing(_ref) {
  var percent = _ref.percent;
  return _react["default"].createElement("div", {
    className: "rainbow-progress-circular_ring"
  }, _react["default"].createElement("svg", {
    xmlns: "http://www.w3.org/2000/svg",
    xmlnsXlink: "http://www.w3.org/1999/xlink",
    version: "1.1",
    width: "140",
    height: "140",
    viewBox: "-16.8 -16.8 33.6 33.6"
  }, _react["default"].createElement("circle", {
    className: "rainbow-progress-circular_ring-path",
    strokeWidth: "1.6",
    fill: "none",
    cx: "0",
    cy: "0",
    r: "16"
  }), _react["default"].createElement("circle", {
    className: "rainbow-progress-circular_ring-fill-percent",
    strokeWidth: "1.6",
    strokeDasharray: "".concat(percent, " ").concat(100 - percent),
    fill: "none",
    cx: "0",
    cy: "0",
    r: "16",
    transform: "rotate(-90)"
  })));
}

ProgressRing.propTypes = {
  percent: _propTypes["default"].number.isRequired
};
