// react-rainbow-components/components/Textarea/label.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = Label;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _classnames = _interopRequireDefault(require("classnames"));

var _RequiredAsterisk = _interopRequireDefault(require("../RequiredAsterisk"));

function Label(props) {
  var label = props.label,
      required = props.required,
      textareaId = props.textareaId,
      id = props.id,
      readOnly = props.readOnly,
      hideLabel = props.hideLabel;

  var getLabelClassNames = function getLabelClassNames() {
    return (0, _classnames["default"])('rainbow-textarea_label', {
      'rainbow-textarea_label--hide-label': hideLabel
    });
  };

  return _react["default"].createElement("label", {
    className: getLabelClassNames(),
    htmlFor: textareaId,
    id: id,
    readOnly: readOnly
  }, _react["default"].createElement(_RequiredAsterisk["default"], {
    required: required
  }), label);
}

Label.propTypes = {
  label: _propTypes["default"].node.isRequired,
  required: _propTypes["default"].bool.isRequired,
  textareaId: _propTypes["default"].string.isRequired,
  id: _propTypes["default"].string,
  readOnly: _propTypes["default"].bool,
  hideLabel: _propTypes["default"].bool
};
Label.defaultProps = {
  id: undefined,
  readOnly: false,
  hideLabel: false
};
