// react-rainbow-components/components/ImportRecordsFlow/stepThree/assignFieldModalFooter.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = AssignFieldModalFooter;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _Button = _interopRequireDefault(require("../../Button"));

function AssignFieldModalFooter(props) {
  var onCancel = props.onCancel,
      onAssign = props.onAssign,
      isAssignButtonDisabled = props.isAssignButtonDisabled;
  return _react["default"].createElement("div", {
    className: "rainbow-import-records-flow_step-three-assign-field-modal-footer"
  }, _react["default"].createElement(_Button["default"], {
    className: "rainbow-import-records-flow_step-three-assign-field-modal-footer_cancel-button",
    label: "Cancel",
    onClick: onCancel
  }), _react["default"].createElement(_Button["default"], {
    label: "Assign",
    variant: "brand",
    onClick: onAssign,
    disabled: isAssignButtonDisabled
  }));
}

AssignFieldModalFooter.propTypes = {
  onCancel: _propTypes["default"].func,
  onAssign: _propTypes["default"].func,
  isAssignButtonDisabled: _propTypes["default"].bool
};
AssignFieldModalFooter.defaultProps = {
  onCancel: function onCancel() {},
  onAssign: function onAssign() {},
  isAssignButtonDisabled: false
};
