// react-rainbow-components/components/ImportRecordsFlow/stepThree/selectedFieldsToAssign.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = SelectedFieldsToAssign;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _Chip = _interopRequireDefault(require("../../Chip"));

function SelectedFieldsToAssign(_ref) {
  var values = _ref.values,
      _onDelete = _ref.onDelete;
  return values.map(function (value, index) {
    var key = "field-".concat(index);
    return _react["default"].createElement(_Chip["default"], {
      key: key,
      className: "rainbow-import-records-flow_step-three-chip-field",
      label: value,
      title: value,
      variant: "neutral",
      onDelete: function onDelete() {
        return _onDelete(value);
      }
    });
  });
}

SelectedFieldsToAssign.propTypes = {
  values: _propTypes["default"].array,
  onDelete: _propTypes["default"].func
};
SelectedFieldsToAssign.defaultProps = {
  values: [],
  onDelete: function onDelete() {}
};
