// react-rainbow-components/components/Input/inputBase/label.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = Label;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _classnames = _interopRequireDefault(require("classnames"));

var _RequiredAsterisk = _interopRequireDefault(require("../../RequiredAsterisk"));

function Label(props) {
  var label = props.label,
      required = props.required,
      inputId = props.inputId,
      readOnly = props.readOnly,
      id = props.id,
      hideLabel = props.hideLabel;

  var getLabelClassNames = function getLabelClassNames() {
    return (0, _classnames["default"])('rainbow-input_label', {
      'rainbow-input_label-read-only': readOnly,
      'rainbow-input_label--hide': hideLabel
    });
  };

  return _react["default"].createElement("label", {
    className: getLabelClassNames(),
    htmlFor: inputId,
    id: id
  }, _react["default"].createElement(_RequiredAsterisk["default"], {
    required: required
  }), label);
}

Label.propTypes = {
  label: _propTypes["default"].node.isRequired,
  required: _propTypes["default"].bool.isRequired,
  inputId: _propTypes["default"].string.isRequired,
  readOnly: _propTypes["default"].bool.isRequired,
  id: _propTypes["default"].string,
  hideLabel: _propTypes["default"].bool
};
Label.defaultProps = {
  id: undefined,
  hideLabel: false
};
