// react-rainbow-components/components/Chip/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = Chip;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _classnames = _interopRequireDefault(require("classnames"));

var _closeIcon = _interopRequireDefault(require("./closeIcon"));

var _ButtonIcon = _interopRequireDefault(require("./../ButtonIcon"));

var _RenderIf = _interopRequireDefault(require("../RenderIf"));

require("./styles.css");

/**
 * A Chip displays a label that can be removed from view.
 */
function Chip(props) {
  var label = props.label,
      onDelete = props.onDelete,
      variant = props.variant,
      title = props.title,
      className = props.className,
      style = props.style;

  var getVariantClassNames = function getVariantClassNames() {
    if (variant === 'base') {
      return null;
    }

    return "rainbow-chip--".concat(variant);
  };

  var getContainerClassName = function getContainerClassName() {
    return (0, _classnames["default"])('rainbow-chip', getVariantClassNames(), className);
  };

  return _react["default"].createElement("span", {
    className: getContainerClassName(),
    style: style,
    title: title
  }, _react["default"].createElement("span", {
    className: "rainbow-chip_content--truncate"
  }, label), _react["default"].createElement(_RenderIf["default"], {
    isTrue: !!onDelete
  }, _react["default"].createElement(_ButtonIcon["default"], {
    className: "rainbow-chip_button-container",
    icon: _react["default"].createElement(_closeIcon["default"], null),
    size: "small",
    title: "Close",
    onClick: onDelete,
    assistiveText: "Remove"
  })));
}

Chip.propTypes = {
  /** The content to be displayed inside the Chip. */
  label: _propTypes["default"].oneOfType([_propTypes["default"].string, _propTypes["default"].node]),

  /** Displays tooltip text when the mouse moves over the element. */
  title: _propTypes["default"].string,

  /** The variant changes the appearance of the Chip. Accepted variants include base,
   * neutral, outline-brand and brand. This value defaults to base. */
  variant: _propTypes["default"].oneOf(['base', 'neutral', 'outline-brand', 'brand']),

  /** The action triggered when the close button is clicked. */
  onDelete: _propTypes["default"].func,

  /** A CSS class for the outer element, in addition to the component's base classes. */
  className: _propTypes["default"].string,

  /** An object with custom style applied to the outer element. */
  style: _propTypes["default"].object
};
Chip.defaultProps = {
  label: null,
  title: undefined,
  variant: 'base',
  onDelete: undefined,
  className: undefined,
  style: undefined
};
